pkg/common/common.go

```diff
@@ -239,3 +240,46 @@ func GetImagePullPolicy(image string) (corev1.PullPolicy, error) {
 	}
 	return corev1.PullAlways, nil
 }
+
+// GenerateCliArgsFromConfigMap generates CLI arguments from a ConfigMap.
+//
+// This function takes a CLI subcommand and a ConfigMap, and returns a slice of strings representing
+// the CLI arguments from the ConfigMap.
+// The function processes each key-value pair in the ConfigMap as follows:
+//
+//  1. If the ConfigMaps' value starts and ends with single (') or double (") quotes, it is left unchanged.
+//  2. If the ConfigMaps' key starts with single '-' or double '--', it is left unchanged.
+//  3. If an argument key is passed without `-` or `--` prefix, then `--` is added as a key prefix.
+//  4. If the ConfigMap value is "true" or "false" (case-insensitive), it is converted to lowercase
+//     and used without single quotes surroundings (boolean value).
+//  5. The formatted key-value pair is added to the result.
+//
+// Args:
+//
+//	cliSubCommand: The CLI subcommand as a string, for example 'server'
+//	configMap: A pointer to a corev1.ConfigMap containing key-value pairs.
+//
+// Returns:
+//
+//	A slice of strings representing the CLI arguments.
+func GenerateCliArgsFromConfigMap(cliSubCommand string, configMap *corev1.ConfigMap) []string {
```

Comment:
I think you need to modify the function signature here to accept a variable number of string arguments by using variadic parameters. For node-agent daemonset the clisubcommand is `node-agent` and then `server` 
So the function signature should be:
```
func GenerateCliArgsFromConfigMap(cliSubCommand ...string, configMap *corev1.ConfigMap) []string
```

Reply:
Are you sure this is the case?

I do not see another subcommand for the node-agent, just flags.

```shell
$ velero node-agent --help
Work with node-agent

Usage:

Flags:
  -h, --help   help for node-agent

Global Flags:
      --add_dir_header                   If true, adds the file directory to the header of the log messages
      --alsologtostderr                  log to standard error as well as files (no effect when -logtostderr=true)
      --colorized optionalBool           Show colored output in TTY. Overrides 'colorized' value from $HOME/.config/velero/config.json if present. Enabled by default
      --features stringArray             Comma-separated list of features to enable for this Velero process. Combines with values from $HOME/.config/velero/config.json if present
      --kubeconfig string                Path to the kubeconfig file to use to talk to the Kubernetes apiserver. If unset, try the environment variable KUBECONFIG, as well as in-cluster configuration
      --kubecontext string               The context to use to talk to the Kubernetes apiserver. If unset defaults to whatever your current-context is (kubectl config current-context)
      --log_backtrace_at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log_dir string                   If non-empty, write log files in this directory (no effect when -logtostderr=true)
      --log_file string                  If non-empty, use this log file (no effect when -logtostderr=true)
      --log_file_max_size uint           Defines the maximum size a log file can grow to (no effect when -logtostderr=true). Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                      log to standard error instead of files (default true)
  -n, --namespace string                 The namespace in which Velero should operate (default "velero")
      --one_output                       If true, only write logs to their native severity level (vs also writing to each lower severity level; no effect when -logtostderr=true)
      --skip_headers                     If true, avoid header prefixes in the log messages
      --skip_log_headers                 If true, avoid headers when opening log files (no effect when -logtostderr=true)
      --stderrthreshold severity         logs at or above this threshold go to stderr when writing to files and stderr (no effect when -logtostderr=true or -alsologtostderr=false) (default 2)
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging

```

Reply:
@mpryc `velero node-agent server ...` on the node agent vs `velero server ...` on the velero pod

Reply:
Updated the function to allow multiple artguments (here the order in which they are passed to the function matters, so `"node-agent", "server"` will be different from `"server", "node-agent"`) + added sorting of flags so the result is consistent across different function calls.